curvefs/src/client/vfs/operations.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ *  Copyright (c) 2023 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: Curve
+ * Created Date: 2023-07-07
+ * Author: Jingli Chen (Wine93)
+ */
+
+#include <vector>
+
+#include "curvefs/src/client/sdk_helper.h"
+#include "curvefs/src/client/vfs/meta.h"
+#include "curvefs/src/client/vfs/operations.h"
+
+namespace curvefs {
+namespace client {
+namespace vfs {
+
+using ::curvefs::client::SDKHelper;
+
+OperationsImpl::OperationsImpl(std::shared_ptr<Permission> permission,
+                               std::shared_ptr<FuseClient> client)
+    : permission_(permission),
+      client_(client) {}
+
+std::shared_ptr<FuseContext> OperationsImpl::NewFuseContext() {
+    auto uid = permission_->GetCurrentUid();
+    auto gid = permission_->GetCurrentGid();
+    auto umask = permission_->GetCurrentUmask();
+    return std::make_shared<FuseContext>(uid, gid, umask);
+}
+
+// init
+CURVEFS_ERROR OperationsImpl::Mount(const std::string& fsname,
+                                    const std::string& mountpoint,
+                                    FuseClientOption option) {
+    auto helper = SDKHelper();
+    return helper.Mount(client_, fsname, mountpoint, option);
+}
+
+CURVEFS_ERROR OperationsImpl::Umount(const std::string& fsname,
+                                     const std::string& mountpoint) {
+    auto helper = SDKHelper();
+    return helper.Umount(client_, fsname, mountpoint);
+}
+
+// directory*
+CURVEFS_ERROR OperationsImpl::MkDir(Ino parent,
+                                    const std::string& name,
+                                    uint16_t mode,
+                                    EntryOut* entryOut) {
+    auto ctx = NewFuseContext();
```

Comment:
Is it possible to use FuseContext as a variable instead of creating a new one every time?

Reply:
> Is it possible to use FuseContext as a variable instead of creating a new one every time?

No, because we should get the latest `uid` and `gid` which can modified on fly.